xla/service/gpu/gpu_hlo_cost_analysis.h

```diff
@@ -47,6 +50,8 @@ class GpuHloCostAnalysis : public HloCostAnalysis {
   Status HandleElementwiseUnary(const HloInstruction* hlo) override;
   Status HandleElementwiseBinary(const HloInstruction* hlo) override;
 
+  Status HandleAllReduce(const HloInstruction* hlo) override;
```

Comment:
Looks like this is a subclass of ConstDfsHloVisitor. So don't we need to override HandleAllReduceStart for use in LHS (as all collectives are converted to their async form?)

Reply:
HandleAllReduceStart is overriden in HloAnalysis class to call HandleAllReduce, so we just need to override HandleAllReduce here